src/transformers/image_box_utils.py

```diff
@@ -0,0 +1,523 @@
+        raise ValueError(
+            f"Expected boxes to have at most 3 dimensions (n_images, n_boxes, coords), got {depth} dimensions."
+        )
+
+    # check image size is provided correctly
+    if depth == 3 and image_size is not None:
+        image_depth = get_depth_of_nested_objects(image_size)
+        if image_depth == 1:
+            raise ValueError(f"Expected get list of image_sizes but get a single image size: {image_size}.")
+        elif len(image_size) != len(boxes):
+            raise ValueError(
+                f"Expected the same number of images and image sizes, got {len(boxes)} images and {len(image_size)} image sizes."
+            )
+    if depth < 3 and image_size is not None and len(image_size) != 2:
+        raise ValueError(f"Expected image_size to have 2 elements (height, width), got {len(image_size)} elements.")
+
+    # check box has enough coordinates
+    if depth == 1 and len(boxes) < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {len(boxes)} elements.")
+    if is_array_type(boxes) and _shape(boxes)[-1] < 4:
+        raise ValueError(f"Expected boxes to have at least 4 elements, got {_shape(boxes)[-1]} elements.")
+
+    # Base of recursion.
+    if depth == 2 and is_array_type(boxes):
+        return _convert_boxes_arrays_2d(boxes, input_format, output_format, image_size, check)
+
+    # Recursive approach.
+    elif depth == 1 and isinstance(boxes, (list, tuple) or is_array_type(boxes)):
+        boxes_2d = _expand_on_zero_dim(boxes)
+        return _convert_boxes_recursively(boxes_2d, input_format, output_format, image_size, check)[0]
+
+    elif depth == 2 and isinstance(boxes, (list, tuple)):
+        np_boxes = np.array(boxes)
+        np_converted_boxes = _convert_boxes_recursively(np_boxes, input_format, output_format, image_size, check)
+        converted_boxes = np_converted_boxes.tolist()
+        if isinstance(boxes, tuple):
+            converted_boxes = tuple([tuple(box) for box in converted_boxes])
+        return converted_boxes
+
+    elif depth == 3 and isinstance(boxes, (list, tuple)):
+        image_sizes = image_size if image_size is not None else [None] * len(boxes)
+        return type(boxes)(
+            [_convert_boxes_recursively(b, input_format, output_format, s, check) for b, s in zip(boxes, image_sizes)]
+        )
+
+    elif depth == 3 and is_array_type(boxes):
+        image_sizes = image_size if image_size is not None else [None] * len(boxes)
+        converted_boxes = [
+            _convert_boxes_recursively(b, input_format, output_format, s, check) for b, s in zip(boxes, image_sizes)
+        ]
+        return _stack(converted_boxes, axis=0)
+
+    else:
+        raise ValueError(f"Unsupported type of boxes: {type(boxes)}")
+
+
+def convert_boxes(
+    boxes: Union[ArrayType, List, Tuple],
+    input_format: str,
+    output_format: str,
+    image_size: Optional[Union[ArrayType, List, Tuple]] = None,
+    check: Optional[str] = "warn",
```

Comment:
I can see `"raise"` never being used. I like the logic, but think we might be engineering something which we'll never need